### PR TITLE
Document UXP Versioning Strategy

### LINKED ISSUE #134: Document versioning strategy and LTS
### What problem are you facing?

The original UXP design doc explains the [versioning format for UXP releases](https://github.com/upbound/universal-crossplane/blob/main/design/design-packaging.md#versioning). but does not give explicit direction as to when tags can be cut, which branches they can be applied to, and what the policy is for long term support.

### How could Upbound help solve your problem?

Create a formal document that describes all of the aforementioned information. A nice follow-up would be to codify those practices in our repository machinery.

### PR DESCRIPTION
**Pull request by hasheddan**
### Description of your changes

Adds a document describing UXP versioning and release strategy. Does not define LTS strategy as it is still being formalized by UXP maintainers.

Fixes #134 

I have:

- [x] Read and followed Upbound's [contribution process](https://git.io/fj2m9).
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR, as appropriate.

### How has this code been tested

n/a

### CODE REVIEW

File: VERSIONING.md
Comment (AaronME) [inline VERSIONING.md:31]:
```suggestion
precedence is honored in semantic versioning. The `<components>` numeric
```

File: VERSIONING.md
Comment (AaronME) [inline VERSIONING.md:34]: NB, for absolute clarity:

```suggestion
`<components>` versions across `<major>`, `<minor>`, `<patch>` versions. (For example, "1.2.0-up.1" does not indicate any correspondence of components with "1.3.0-up.1.")
```

**Review by AaronME — APPROVED** [review]

File: VERSIONING.md
Comment (turkenh) [inline VERSIONING.md:87]: Would this mean we will be tagging the same commit first with the released version and then after the release with the RC version? Does our build tooling handle multiple tags on the same commit properly (e.g. use the latest applied tag) or would we need a dummy commit to move the head of the release branch before applying RC tag?

File: VERSIONING.md
Comment (turkenh) [inline VERSIONING.md:94]: `rc-` starts with `0` but `up-` starts with `1`. Could we follow the same for both?

File: VERSIONING.md
Comment (hasheddan) [reply VERSIONING.md:24]: Good question -- We don't want any temporary fork to impact upstream release versioning (nor do we really have the power to do that). I imagine in this case that we would keep the Crossplane version the same and bump the `-up.X` version. This would allow for the next upstream patch release to be consumed with the presumed bugfix and for us to bump to the appropriate Crossplane version.

File: VERSIONING.md
Comment (hasheddan) [reply VERSIONING.md:87]: Interestingly our build tooling does handle tags on the same commit (it will use the most recent by timestamp), but I am an advocate for merging an empty commit and tagging it instead. This is how we advance branches upstream (ref https://github.com/crossplane/crossplane/pull/2395).

File: VERSIONING.md
Comment (hasheddan) [reply VERSIONING.md:94]: I would be in favor of that, but I was basing this off of https://github.com/upbound/universal-crossplane/blob/main/design/design-packaging.md#versioning, in which I feel like there was a reason for starting with `-up.1`, but I cannot remember why, do you happen to know?

File: VERSIONING.md
Comment (negz) [inline VERSIONING.md:49]: I don't disagree with this practice, or have any better ideas, but I am a little worried that it will be very easy to mix up an RC version of a UXP release (e.g. `v1.3.0-up.1-rc.0`) and a UXP release of an upstream RC version (e.g. `v1.3.0-rc.0-up.1`).

File: VERSIONING.md
Comment (negz) [inline VERSIONING.md:72]: Doesn't this counter the statement above?

> Additional pre-release segments may be appended both before and after the `-up-<components>` segment.

Comment (negz) [reply VERSIONING.md:72]: Oh - I think I missed the key point that this is talking about the main branch specifically.

File: VERSIONING.md
Comment (negz) [inline VERSIONING.md:106]: Not a priority, but this seems like something we could automate pretty easily (e.g. with a cronjob style GitHub Action on the UXP repo).

**Review by negz — APPROVED** [review]
I think there's a few small threads to close, and I'd like to get sign-off from @turkenh and @luebken before we merge, but this broadly LGTM.

File: VERSIONING.md
Comment (negz) [reply VERSIONING.md:49]: One fairly gross way to clarify would be to use something like `uprc` to distinguish UXP release candidates. e.g. `v1.3.0-up.1-uprc.0`. 😬

File: VERSIONING.md
Comment (hasheddan) [reply VERSIONING.md:106]: @negz totally agree :) it was in the original doc (as suggested by @muvaf), but didn't include here since this will be a long-lived doc. I'll make sure to open some issues after merge 👍🏻

File: VERSIONING.md
Comment (hasheddan) [reply VERSIONING.md:49]: I am not strongly opposed to this and don't think it takes us any farther away from conventions than we already are. Happy to go either way on this -- @turkenh do you have thoughts?

File: VERSIONING.md
Comment (turkenh) [reply VERSIONING.md:87]: Do you think if it makes sense to document this here?

File: VERSIONING.md
Comment (turkenh) [reply VERSIONING.md:94]: No, I don't remember either. I believe we can start with `-up.0` similar to how a patch starts with 0, e.g. `1.3.0`.

File: VERSIONING.md
Comment (turkenh) [reply VERSIONING.md:49]: I am still wondering if we could completely drop off `-up.1` in UXP main as I commented before :)

**Review by turkenh — APPROVED** [review]
Nothing blocking, lgtm!
Thanks @hasheddan.

File: VERSIONING.md
Comment (hasheddan) [reply VERSIONING.md:87]: Done 👍🏻

File: VERSIONING.md
Comment (hasheddan) [reply VERSIONING.md:49]: @turkenh IMO that would break our versioning format because we are essentially extending our semantic version specification. I would equate it with tagging a Crossplane version with `v1.2-rc.0` instead of `v1.2.0-rc.0` on `master` (though I do acknowledge this is not an exact parallel 🙂 ). Because these will only be RC versions anyway, let's start off with keeping the `-up.X` version for now and we can change course in the future if needed 👍🏻 

Also, I decided to add the `uprc` change 👍🏻

**Pull request merged**
